backend/inspirehep/matcher/tasks.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from celery import shared_task
+from invenio_db import db
+from invenio_records.api import RecordMetadata
+from sqlalchemy import type_coerce
+from sqlalchemy.dialects.postgresql import JSONB
+
+from inspirehep.records.api import LiteratureRecord
+
+from .api import match_references
+
+
+@shared_task(
+    ignore_result=False, queue="matcher",
+)
+def match_references_by_uuids(literature_uuids):
+    # TODO: do not crash all if it fails all record
+    record_json = type_coerce(RecordMetadata.json, JSONB)
+    has_references = record_json.has_key("references")  # noqa: W601
+    selected_uuids = RecordMetadata.id.in_(literature_uuids)
+    with_references_query = RecordMetadata.query.filter(selected_uuids, has_references)
+    for record_metadata in with_references_query.all():
+        references = record_metadata.json.get("references")
+        matched_references = match_references(references)
+        if references != matched_references:
```

Comment:
I think you want
```suggestion
        if references is not matched_references:
```
to avoid doing the deep equality check.